crates/rethnet_eth/src/remote/client.rs

```diff
@@ -141,36 +184,76 @@ impl RpcClient {
             .create(directory.clone())
             .await
             .map_err(|error| RpcClientError::CacheError {
-                message: "failed to create on-disk RPC response cache".to_string(),
-                error,
+                message: "failed to create RPC response cache".to_string(),
+                cache_key: cache_key.to_string(),
+                error: error.into(),
             })?;
 
         let path = Path::new(&directory).join(format!("{}.json", cache_key));
         Ok(path)
     }
 
+    /// Don't fail the request, just log an error if we fail to read from cache.
+    fn log_cache_error(cache_key: CacheKey, message: &'static str, error: impl Into<CacheError>) {
```

Comment:
As this is not linked to the struct, make it a free function within the file.

I follow this heuristic: https://drdobbs.com/cpp/how-non-member-functions-improve-encapsu/184401197

Reply:
Thanks, good call! Fixed in [9664ac8](https://github.com/NomicFoundation/hardhat/pull/4315/commits/9664ac82d6170abef9ced65f23d7722c24b87b73)